src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -396,12 +396,14 @@ t8_forest_element_from_ref_coords_ext (t8_forest_t forest, t8_locidx_t ltreeid,
                                        const double *ref_coords, const size_t num_coords, double *coords_out,
                                        const double *stretch_factors)
 {
-  double tree_ref_coords[3] = { 0 };
   const t8_eclass_t tree_class = t8_forest_get_tree_class (forest, ltreeid);
+  const int tree_dim = t8_eclass_to_dimension[tree_class];
   const t8_eclass_scheme_c *scheme = t8_forest_get_eclass_scheme (forest, tree_class);
   const t8_cmesh_t cmesh = t8_forest_get_cmesh (forest);
   const t8_gloidx_t gtreeid = t8_forest_global_tree_id (forest, ltreeid);
 
+  double *tree_ref_coords = T8_ALLOC (double, (tree_dim == 0 ? 1 : tree_dim) * num_coords);
```

Comment:
@jfussbro  Please answer this comment.

Reply:
No, because the computation for 2d elements only output 2 dimensional coords. For example quads:
https://github.com/DLR-AMR/t8code/blob/a315a1f1d00c43f79b30f49a88cef7eb5b3e849d/src/t8_schemes/t8_default/t8_default_quad/t8_dquad_bits.c#L26-L42

Reply:
> The triangle scheme, for example, assumes three-component vector:
> 
> https://github.com/DLR-AMR/t8code/blob/0405df308813e87b14dbe42224badd672ff5a4ae/src/t8_schemes/t8_default/t8_default_tri/t8_dtri_bits.c#L429

Yes, but only when the `T8_DTRI_TO_DTET` Macro is active. Otherwise it requires only a 2d vector.